packages/esm-patient-chart-app/src/visit/visit-form/visit-attribute-type.component.tsx

```diff
@@ -71,13 +61,11 @@ const VisitAttributeTypeFields: React.FC<VisitAttributeTypeFieldsProps> = ({
 };
 
 interface AttributeTypeFieldProps {
+  fieldProps: ControllerRenderProps<VisitFormData, `visitAttributes.${string}`>;
```

Comment:
Shouldn't this be `FieldPath<VisitFormData>` or similar?

Reply:
This is the type of the `field`.

![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/51502471/9ba33dcd-ffed-4a09-a285-35285e1ed5c6)

Reply:
![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/51502471/ed7afc26-6a6a-4124-ac26-8580da8e68f4)

Reply:
Right, because we've told it that the property is `visitAttributes.${attributeType.uuid}`, which works at runtime, but not at the level of the type system, since the type system only exists when adding or editing code, so it infers `attributeType.uuid` is a type string and seems to substitute that.

Reply:
![image](https://github.com/openmrs/openmrs-esm-patient-chart/assets/51502471/d546a523-5e0c-4d2d-bd7e-2eac77bd10a1)


I'm not able to understand this, with my understanding, the second arg `visitAttributes.${string}` will be the FieldPath, not the whole `field`, right?